src/chess/uciloop.cc

```diff
@@ -247,10 +250,21 @@ void UciLoop::SendInfo(const std::vector<ThinkingInfo>& infos) {
     if (info.seldepth >= 0) res += " seldepth " + std::to_string(info.seldepth);
     if (info.time >= 0) res += " time " + std::to_string(info.time);
     if (info.nodes >= 0) res += " nodes " + std::to_string(info.nodes);
-    if (info.score) res += " score cp " + std::to_string(*info.score);
+    // Mate display. If abs(score) > 20000 its a certain win (or loss).
+    // Length to mate (or tbwin) is then (+ or -) abs(score)-20000. 
+    if (info.score) {
```

Comment:
It's not nice at all to reuse `info.score` for checkmate display. Add `info.mate` for that.

Reply:
Yes, I know. Will fix. Will do this last, when everything else is cleaned up. We would still need an indicator if info.score or info.mate is valid when processing the info structure. Ideas?